server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -70,14 +70,26 @@ public Map<String, Long> toBreakdownMap() {
         );
         final long createWeightTime = topLevelBreakdownMapWithWeightTime.get(QueryTimingType.CREATE_WEIGHT.toString());
 
-        if (sliceCollectorsToLeaves.isEmpty() || contexts.isEmpty()) {
+        if (contexts.isEmpty()) {
             // If there are no leaf contexts, then return the default concurrent query level breakdown, which will include the
             // create_weight time/count
             queryNodeTime = createWeightTime;
             maxSliceNodeTime = 0L;
             minSliceNodeTime = 0L;
             avgSliceNodeTime = 0L;
             return buildDefaultQueryBreakdownMap(createWeightTime);
+        } else if (sliceCollectorsToLeaves.isEmpty()) {
+            assert contexts.size() == 1 : "Unknown cases";
```

Comment:
Can you add a comment after the `else if` statement:
`This will happen when each slice executes search leaf for its leaves and query is rewritten for the leaf being searched. It creates a new weight and breakdown map for each rewritten query. This new breakdown map captures the timing information for the new rewritten query. The sliceCollectorsToLeaves is empty because this breakdown for rewritten query gets created later in search leaf path which doesn't have collector. Also this is not needed since this breakdown is per leaf and there is no concurrency involved.`

Reply:
Hi @sohami, thanks for the detailed comment; it has now been addressed.